### PR TITLE
Task/gp 790 delete vs item

### PR DESCRIPTION
**Pull request by danielhedlund**
## What does this change?

Implements the second pillar of `MediaNotRequiredProcessor`, namely handling removal of media from the online (Vidispine) storage.

Obviously, there are several checks to see if it is appropriate to actually delete a media item, including checks of project state, media existing on nearline, internal archive or deep archive if needed, etc.

## Why is the change needed?

Remove media from online, to lessen space used.

## Important implementation details
- **Getting the MD5 of online (vidispine) media by parsing also metadata in `VSShapeFile` (NB **_not_** added to `FileDocument`)**
- Reworked looking up `PendingDeletionRecord`s, as filepath is not unique (because of versioning)
- Extracted methods from `performActionNearline` 
  - easier to see logic
  - some also usable by `performActionOnline`

**Still missing**
* Self-healing loop when media doesn't exists where it needs to exist for backup purposes
* ~Not yet getting the MD5 checksum of online files to use in check for matching files - currently relying on filepath + filesize solely.~

### CODE REVIEW

File: media_remover/src/main/scala/MediaNotRequiredMessageProcessor.scala [hunk 1]
```diff
@@ -424,29 +409,34 @@ class MediaNotRequiredMessageProcessor(asLookup: AssetFolderLookup)(
       } yield maybeMd5
   }
 
-  def NOT_IMPL_getChecksumForOnline(vsItemId: String): Future[Option[String]] = {
-    // TODO? How to get the -- md5 -- checksum for a vidispine item? VS defaults to SHA-1 unless overridden in configuration property 'fileHashAlgorithm'
-    ???
+
+  def STUB_ALWAYS_RETURNS_NONE_getChecksumForOnline(vsItemId: String): Future[Option[String]] = {
+    // How to get the -- md5 -- checksum for a vidispine item? VS defaults to SHA-1 unless overridden in configuration property 'fileHashAlgorithm'
+    // FIXME Read the value of FileDocument/metadata/field/key/hash-MD5
```
Comment (fredex42) [inline media_remover/src/main/scala/MediaNotRequiredMessageProcessor.scala:415]: As you probably saw, we now have optional MD5 checksumming enabled.

File: media_remover/src/main/scala/MediaNotRequiredMessageProcessor.scala [hunk 2]
```diff
@@ -470,29 +460,40 @@ class MediaNotRequiredMessageProcessor(asLookup: AssetFolderLookup)(
     })
   }
 
+
   def removeDeletionPending(existingRecord: PendingDeletionRecord): Future[Int] =
     pendingDeletionRecordDAO.deleteRecord(existingRecord)
 
+
   def removeDeletionPendingByMessage(msg: OnlineOutputMessage): Future[Either[String, Int]] =
```
Comment (fredex42) [inline media_remover/src/main/scala/MediaNotRequiredMessageProcessor.scala:468]: What's the `Right()` return value here?

File: media_remover/src/main/scala/MediaNotRequiredMessageProcessor.scala (hunk 1 above)
Comment (danielhedlund) [reply media_remover/src/main/scala/MediaNotRequiredMessageProcessor.scala:415]: Yes, working on it right now.

File: media_remover/src/test/scala/MediaNotRequiredMessageProcessorSpec.scala [hunk 3]
```diff
@@ -1,13 +1,16 @@
+import MediaNotRequiredMessageProcessor.Action
 import akka.actor.ActorSystem
+import akka.http.javadsl.model.HttpMessage.DiscardedEntity
```
Comment (fredex42) [inline media_remover/src/test/scala/MediaNotRequiredMessageProcessorSpec.scala:3]: Might be an erroneous import? Normally would expect to see `akka.http.scaladsl` not `javadsl`

**Review by fredex42 — COMMENTED** [review]
Looking really great, couple of inline queries but 👍

File: media_remover/src/test/scala/MediaNotRequiredMessageProcessorSpec.scala (hunk 3 above)
Comment (danielhedlund) [reply media_remover/src/test/scala/MediaNotRequiredMessageProcessorSpec.scala:3]: Ah, yes.

File: media_remover/src/main/scala/MediaNotRequiredMessageProcessor.scala (hunk 2 above)
Comment (danielhedlund) [reply media_remover/src/main/scala/MediaNotRequiredMessageProcessor.scala:477]: Sure!

**Pull request closed without merging**